pyvista/plotting/camera.py

```diff
@@ -690,23 +705,36 @@ def tight(self, padding=0.0, adjust_render_window=True):
 
         self._renderer.ComputeAspect()
         aspect = self._renderer.GetAspect()
-        angle = np.pi * self.view_angle / 180.0
-        dx, dy = (x1 - x0), (y1 - y0)
-        dist = max(dx / aspect[0], dy) / np.sin(angle / 2) / 2
 
-        self.SetViewUp(0, 1, 0)
-        self.SetPosition(x0 + dx / 2, y0 + dy / 2, dist * (1 + padding))
-        self.SetFocalPoint(x0 + dx / 2, y0 + dy / 2, 0)
+        position0 = np.array([x0, y0, z0])
+        position1 = np.array([x1, y1, z1])
+        objects_size = position1 - position0
+        position = position0 + objects_size / 2
+
+        direction, viewup = view_vectors(view, negative)
+        horizontal = np.cross(direction, viewup)
+
+        vert_dist = np.abs(np.dot(objects_size, viewup))
+        horiz_dist = np.abs(np.dot(objects_size, horizontal))
+
+        # set focal point to objects center
```

Comment:
Very tiny nitpick
```suggestion
        # set focal point to objects' center
```
or
```suggestion
        # set focal point to object's center
```
Which brings me to a question: does this also work when there are multiple meshes in the scene? How is this center computed?

Reply:
I added a test for this.  It uses the bounds of all the viewable objects in the scene and determines the 'view center'.